### DISCUSSION TITLE
Tons of timeouts in local network without internet / cloud connection

### DISCUSSION THREAD

**Discussion by pustelbaer** [smarthausleben/ioBroker.wamo#29] [General]
Last week my internet router died and my smart home was running without cloud connections. My SYR SafeTech was still reachable under the usual IP address, because the DHCP server runs locally on my NAS. Nevertheless, the communication to the SafeTech did not work as usual anymore. For some reason commands and queries are not always executed. For example, switching a profile no longer works reliably. Often the command has to be sent several times until it is executed. It would have to be clarified whether this is a problem of the SafeTech implementation or whether the WAMO adapter has to communicate differently in such a case. In principle, queries and commands still seem to work, but certain operations seem to take longer or cause a timeout, which hinders the actual communication. 
I think it would make sense to implement a retry mechanism in WAMO. When a command is sent it should be applied. The SafeTech doesn't seem to be very responsive, so the adapter should resend certain operations if they are not executed. I am thinking in particular of switching a profile. It is very annoying when the SafeTech trips because the absence profile is still active even though the presence profile was requested.
Here is an sample excerpt from the log. Timeouts occur repeatedly.

![grafik](https://github.com/smarthausleben/ioBroker.wamo/assets/35026034/1502f461-7c84-4393-88f5-56d43315ff4a)

**Comment by smarthausleben** [comment]
I don’t know how the device behaves if it has no cloud connection. But in my opinion it shouldn’t disturb the local communication so much.

But to verify if a given command went thru is relatively easy. You need just to check the specific parameter, after you changed it, if it has the desired value and if it is acknowledged. If not, just send the command again until it went thru.
